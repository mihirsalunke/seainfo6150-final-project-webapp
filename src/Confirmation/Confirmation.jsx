import React from "react";
import PropTypes from "prop-types";
import styles from "./Confirmation.module.css";

const Confirmation = (props) => {
    return (
        <div className={styles.container}>
            <div className={styles.confimationMessage}>
                <p>Congratulations..!! <br/>You are registered for <b>{props.event.title}</b></p>
            </div>
            
            <div className={styles.eventDetails}>
                <div className={styles.title}>
                    <p>Event Details:</p>
                </div>
                
                <div className={styles.datetime}>
                    <p>Event Date: <time dateTime={props.event.date}>{props.event.eventDate}</time></p>
                    <p>Event Time: <time dateTime={props.event.time}>{props.event.eventTime}</time></p>
                </div>
                <div className={styles.location}>
                    <p>Event Location: <span>{props.event.location}</span></p>
                </div>
            </div>

            <div className={styles.participantDetails}>
                <div className={styles.title}>
                    <p>Participant's Details:</p>
                </div>

                <div className={styles.text}>
                    <p>Participant Name: <span>{props.name}</span></p>
                </div>
                <div className={styles.text}>
                    <p>Contact Number: <span>{props.contactNumber}</span></p>
                </div>
                <div className={styles.text}>
                    <p>Email: <span>{props.email}</span></p>
                </div>
            </div>
        </div>
    );
};

Confirmation.propTypes = {
    event: PropTypes.object.isRequired,
    name: PropTypes.string.isRequired,
    contactNumber: PropTypes.string.isRequired,
    email: PropTypes.string.isRequired,
};

export default Confirmation;