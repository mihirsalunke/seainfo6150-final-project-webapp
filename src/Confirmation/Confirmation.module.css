@import url("https://fonts.googleapis.com/css2?family=Rowdies:wght@300&display=swap");
@import url("https://fonts.googleapis.com/css2?family=Nerko+One&display=swap");

.container {
    background: white;
    min-width: 700px;
    border: 1px solid black;
    display: flex;
    flex-direction: column;
    justify-content: center;
    box-shadow: 0px 0px 10px 5px;
    position: fixed;
    top: 40%;
    left: 50%;
    transform: translate(-50%, -40%);
    border-radius: 10px;
}

.confimationMessage {
    font-size: 2em;
    margin: 2%;
    color: rgb(33, 141, 6);
    text-align: center;
}

.confimationMessage b {
    font-family: Rowdies, sans-serif;
}

.eventDetails {
    border: 1px solid black;
    border-radius: 7px;
    margin: 2%;
    
}

.title {
    text-align: center;
    font-family: Rowdies, sans-serif;
}

.datetime {
    padding-left: 5%;
}

.datetime time{
    font-family: "Nerko One", sans-serif;
}

.location {
    padding-left: 5%;
}

.location span {
    font-family: "Nerko One", sans-serif;
}

.participantDetails {
    border: 1px solid black;
    border-radius: 7px;
    margin: 2%;
}

.text {
    padding-left: 5%;
}