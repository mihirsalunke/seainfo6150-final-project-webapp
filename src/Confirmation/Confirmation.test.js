import React from "react";
import Confirmation from "./Confirmation.jsx";
import { render } from "@testing-library/react";

const event = {
    "eventId": "1110001",
    "title": "Future of AI",
    "description": "Future of AI is showcasing latest innovation and business modules",
    "image" : "https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcS9tSKBQBaTBTW3fMOh_8KPz5CqGeT6AeptUA&usqp=CAU",
    "location": "Chicago",
    "organizerName": "B2C Tech Inc",
    "organizerContactNumber": "351-800-0120",
    "category": "Tech",
    "date": "2020-12-24",
    "time": "13:00",
    "eventDate": "December 24th 2020",
    "eventTime": "1:00 pm"
};

const name = "Mihir";
const contactNumber = "123-456-7890";
const email = "abc@xyz.com";

describe("Confirmation tests", () => {
    it("renders correctly", () => {
      const { container } = render(<Confirmation event={event} name={name} contactNumber={contactNumber} email={email} />);
      expect(container).toMatchSnapshot();
    });
});
