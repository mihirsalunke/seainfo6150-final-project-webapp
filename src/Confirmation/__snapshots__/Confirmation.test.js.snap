// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Confirmation tests renders correctly 1`] = `
<div>
  <div
    class="container"
  >
    <div
      class="confimationMessage"
    >
      <p>
        Congratulations..!! 
        <br />
        You are registered for 
        <b>
          Future of AI
        </b>
      </p>
    </div>
    <div
      class="eventDetails"
    >
      <div
        class="title"
      >
        <p>
          Event Details:
        </p>
      </div>
      <div
        class="datetime"
      >
        <p>
          Event Date: 
          <time
            datetime="2020-12-24"
          >
            December 24th 2020
          </time>
        </p>
        <p>
          Event Time: 
          <time
            datetime="13:00"
          >
            1:00 pm
          </time>
        </p>
      </div>
      <div
        class="location"
      >
        <p>
          Event Location: 
          <span>
            Chicago
          </span>
        </p>
      </div>
    </div>
    <div
      class="participantDetails"
    >
      <div
        class="title"
      >
        <p>
          Participant's Details:
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Participant Name: 
          <span>
            Mihir
          </span>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Contact Number: 
          <span>
            123-456-7890
          </span>
        </p>
      </div>
      <div
        class="text"
      >
        <p>
          Email: 
          <span>
            abc@xyz.com
          </span>
        </p>
      </div>
    </div>
  </div>
</div>
`;
